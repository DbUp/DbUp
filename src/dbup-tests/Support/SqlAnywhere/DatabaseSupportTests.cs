#if !NETCORE
using DbUp.Builder;
using DbUp.SqlAnywhere;
using DbUp.Tests.Common;

namespace DbUp.Tests.Providers.SqlAnywhere;

public class DatabaseSupportTests : DatabaseSupportTestsBase
{
    public DatabaseSupportTests() : base()
    {
    }

    protected override UpgradeEngineBuilder DeployTo(SupportedDatabases to)
        => to.SqlAnywhereDatabase("");

    protected override UpgradeEngineBuilder AddCustomNamedJournalToBuilder(UpgradeEngineBuilder builder, string schema, string tableName)
        => builder.JournalTo(
            (connectionManagerFactory, logFactory)
                => new SqlAnywhereTableJournal(connectionManagerFactory, logFactory, schema, tableName)
        );
}
#endif
