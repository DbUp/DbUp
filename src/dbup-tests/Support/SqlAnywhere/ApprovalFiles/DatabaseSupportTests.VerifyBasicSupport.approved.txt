DB Operation: Open connection
Info:         Beginning database upgrade
Info:         Checking whether journal table exists..
DB Operation: Execute scalar command: select cast(1 as Int) from systab t where t.table_name = 'SCHEMAVERSIONS'
DB Operation: Dispose command
Info:         Journal table does not exist
Info:         Executing Database Server script 'Script0001.sql'
Info:         Checking whether journal table exists..
DB Operation: Execute scalar command: select cast(1 as Int) from systab t where t.table_name = 'SCHEMAVERSIONS'
DB Operation: Dispose command
Info:         Creating the [schemaversions] table
DB Operation: Execute non query command: create table [schemaversions] (
                    [Id] int identity not null constraint [PK_schemaversions_Id] primary key,
                    [ScriptName] nvarchar(255) not null,
                    [Applied] datetime not null
                )
DB Operation: Dispose command
Info:         The [schemaversions] table has been created
DB Operation: Execute non query command: script1contents
DB Operation: Dispose command
DB Operation: Create parameter
Info:         DB Operation: Add parameter to command: scriptName=Script0001.sql
DB Operation: Create parameter
Info:         DB Operation: Add parameter to command: applied=<date>
DB Operation: Execute non query command: insert into [schemaversions] (ScriptName, Applied) values (? ,?)
DB Operation: Dispose command
Info:         Upgrade successful
DB Operation: Dispose connection
